master/buildbot/secrets/manager.py

```diff
@@ -0,0 +1,67 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+"""
+manage vaults and handle secrets
+"""
+from future.utils import text_type
+
+from buildbot.secrets.secret import Secret, ActualSecret
+from buildbot.util.service import BuildbotService
+
+
+class SecretManager(BuildbotService):
+    """
+    Secret manager
+    """
+    name = 'secrets'
+
+    def __init__(self, vaults):
+        super(SecretManager, self).__init__()
+
+        self._vaults = vaults
+
+    def get_vault(self, vault_name):
+        """
+        get vault by its name
+        """
+
+    @classmethod
+    def get_manager(cls, master):
+        """
+        get secret manager provided the given master
+        """
+
+    @classmethod
+    def get(cls, master, secret, *args, **kwargs):
```

Comment:
I think this would make sense to make the VaultManager a first class service, like data and mq, so that you can access it via

    self.master.secrets.get(secret_name)